apex/contrib/torchsched/utils.py

```diff
@@ -0,0 +1,246 @@
+"""General utilities for graph scheduler study."""
+
+from __future__ import annotations
+
+import math
+from typing import Callable
+
+import torch
+import torch.nn as nn
+
+__all__ = ["partition_weighted", "partition_even", "run_and_time"]
+
+cudart = torch.cuda.cudart()
+
+
+def partition_weighted(
+    weights: list[int] | tuple[int, ...],
+    sm_count: int | None = None,
+    rounding_multiple: int = 1,
+) -> list[int]:
+    """Partition the SMs based on their weights, rounding to a specified multiple.
```

Comment:
unclear what "weights" are here, are these GEMM weights?

Reply:
maybe illustrate what the relevant use-case is---are these GEMMs expected to be concurrent (and that's why the sum should be the given `sm_count`?

Reply:
Removed this file since multi-stream doesn't use it.